### ISSUE TITLE
[Windows][OpenCL][Intel] FindOpenCL.cmake module cannot seem to find install OpenCL SDK

### ISSUE THREAD

**Issue by atlury** [BVLC/caffe#5575]
@naibaf7 

Testing OpenCL Caffe in Windows on Core i5-4210U. Installed latest OpenCL SDK, runtime, and drivers. Tried setting OPENCL_LIBRARIES and OPENCL_INCLUDE_DIRS to many combinations, but it doesn't help.

C:\projects\caffe>scripts\build_win.cmd
The system cannot find the drive specified.
INFO: ============================================================
INFO: Summary:
INFO: ============================================================
INFO: MSVC_VERSION               = 14
INFO: WITH_NINJA                 = 0
INFO: CMAKE_GENERATOR            = "Visual Studio 14 2015 Win64"
INFO: CPU_ONLY                   = 0
INFO: USE_CUDA                   = 0
INFO: USE_CUDNN                  = 0
INFO: USE_GREENTEA               = 1
INFO: USE_LIBDNN                 = 1
INFO: USE_OPENMP                 = 0
INFO: USE_INDEX64                =
INFO: USE_INTEL_SPATIAL          = 0
INFO: CMAKE_CONFIG               = Release
INFO: USE_NCCL                   = 0
INFO: CMAKE_BUILD_SHARED_LIBS    = 0
INFO: PYTHON_VERSION             = 3
INFO: BUILD_PYTHON               = 1
INFO: BUILD_PYTHON_LAYER         = 1
INFO: BUILD_MATLAB               = 0
INFO: PYTHON_EXE                 = "python"
INFO: RUN_TESTS                  = 0
INFO: RUN_LINT                   = 0
INFO: RUN_INSTALL                = 0
INFO: ============================================================
-- The C compiler identification is MSVC 19.0.24215.1
-- The CXX compiler identification is MSVC 19.0.24215.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: C:/Program Files/Anaconda3/python.exe (found suitable version "3.5.2", minimum required is "2.7")
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Boost version: 1.61.0
-- Found the following Boost libraries:
--   system
--   thread
--   filesystem
--   chrono
--   date_time
--   atomic
-- Found GFlags: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include
-- Found gflags  (include: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include, library: gflags_shared)
-- Found Glog: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include
-- Found glog    (include: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include, library: glog)
-- Found Protobuf: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/bin/protoc.exe (found version "3.1.0")
-- Found PROTOBUF Compiler: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/bin/protoc.exe
-- Found LMDB: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include
-- Found lmdb    (include: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include, library: lmdb)
-- Found LevelDB: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include
-- Found LevelDB (include: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include, library: leveldb)
-- Found ZLIB: optimized;C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/lib/caffezlib.lib;debug;C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/lib/caffezlibd.lib (found version "1.2.8")
-- Found Snappy: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include
-- Found Snappy  (include: C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/include, library: snappy_static;optimized;C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/lib/caffezlib.lib;debug;C:/Users/Bill/.caffe/dependencies/libraries_v140_x64_py35_1.1.0/libraries/lib/caffezlibd.lib)
-- -- CUDA is disabled. Building without it...
-- Found ViennaCL include: C:/projects/viennacl-dev
CMake Error at C:/Downloads/cmake/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
  Could NOT find OpenCL (missing: OPENCL_LIBRARIES OPENCL_INCLUDE_DIRS)
Call Stack (most recent call first):
  C:/Downloads/cmake/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindOpenCL.cmake:72 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake/Modules/FindViennaCL.cmake:37 (find_package)
  cmake/Dependencies.cmake:116 (find_package)
  CMakeLists.txt:126 (include)

-- Configuring incomplete, errors occurred!
See also "C:/projects/caffe/build/CMakeFiles/CMakeOutput.log".
See also "C:/projects/caffe/build/CMakeFiles/CMakeError.log".
ERROR: Configure failed

**Comment by atlury** [comment]
Workaround is to manually copy the OpenCL lib and include folders to 
C:\Users\billgates\.caffe\dependencies\libraries_v140_x64_py35_1.1.0\libraries\lib and C:\Users\billgates\.caffe\dependencies\libraries_v140_x64_py35_1.1.0\libraries\include respectively and re-run installation.

**Comment by naibaf7** [comment]
Try again with the updated Makefiles, please.

**Comment by atlury** [comment]
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
OPENCL_INCLUDE_DIRS
   used as include directory in directory C:/projects/caffe/build/CMakeFiles/CMakeTmp
_OPENCL_64_LIBRARIES
    linked by target "cmTC_4177a" in directory C:/projects/caffe/build/CMakeFiles/CMakeTmp

CMake Error: Internal CMake error, TryCompile configure of cmake failed
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
OPENCL_INCLUDE_DIRS
   used as include directory in directory C:/projects/caffe/build/CMakeFiles/CMakeTmp
_OPENCL_32_LIBRARIES
    linked by target "cmTC_ec4e2" in directory C:/projects/caffe/build/CMakeFiles/CMakeTmp

CMake Error: Internal CMake error, TryCompile configure of cmake failed
-- Couldn't link opencl..
CMake Error at C:/Downloads/cmake/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
  Could NOT find OpenCL (missing: OPENCL_LIBRARIES OPENCL_INCLUDE_DIRS)
Call Stack (most recent call first):
  C:/Downloads/cmake/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindOpenCL.cmake:154 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake/Modules/FindViennaCL.cmake:37 (find_package)
  cmake/Dependencies.cmake:116 (find_package)
  CMakeLists.txt:126 (include)


-- Configuring incomplete, errors occurred!
See also "C:/projects/caffe/build/CMakeFiles/CMakeOutput.log".
See also "C:/projects/caffe/build/CMakeFiles/CMakeError.log".
ERROR: Configure failed

**Comment by naibaf7** [comment]
Really odd... I tried again and it finds both AMD SDK and CUDA on my installation with valid OpenCL 64bit.

**Comment by atlury** [comment]
Can you try once from your side with OpenCL from Intel in windows? (if time permits and if life allows it)

**Comment by naibaf7** [comment]
Yeah I'll try to set up Windows on my Intel laptop, but after NIPS conference.

**Comment by bxk-sonavex** [comment]
@naibaf7 
I am facing exactly the same problem, have you had chance to test it?  Is this problem solved?  Thanks!

**Comment by bxk-sonavex** [comment]
@naibaf7 
The tricks (workaround) from @atlury worked for me.
